### PR TITLE
treewide: replace -DCMAKE_BUILD_TYPE in cmakeFlags with cmakeBuildType

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Description of changes

To discourage people to add more occurrences of it.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/phonon/backends/gstreamer.nix [hunk 1]
```diff
@@ -7,14 +7,6 @@ stdenv.mkDerivation rec {
...
   version = "4.10.0";
 
-  meta = with lib; {
-    homepage = "https://phonon.kde.org/";
-    description = "GStreamer backend for Phonon";
-    platforms = platforms.linux;
-    maintainers = with maintainers; [ ttuegel ];
-    license = licenses.lgpl21;
-  };
-
```
Comment (bjornfor) [inline pkgs/development/libraries/phonon/backends/gstreamer.nix:17]: Oops? I don't think removing meta should be part of this commit.

File: pkgs/development/libraries/phonon/backends/gstreamer.nix [hunk 2]
```diff
@@ -34,17 +26,16 @@ stdenv.mkDerivation rec {
...
+  env.NIX_CFLAGS_COMPILE = let
+    gstPluginPaths = lib.makeSearchPathOutput "lib" "/lib/gstreamer-1.0"
+      (with gst_all_1; [
+        gstreamer
+        gst-plugins-base
+        gst-plugins-good
+        gst-plugins-ugly
+        gst-plugins-bad
+        gst-libav
+      ]);
```
Comment (bjornfor) [inline pkgs/development/libraries/phonon/backends/gstreamer.nix:38]: Unrelated whitespace change? (Please remove.)

File: pkgs/development/libraries/phonon/backends/gstreamer.nix [hunk 3]
```diff
@@ -67,7 +58,13 @@ stdenv.mkDerivation rec {
...
-  ];
+  cmakeBuildType = if debug then "Debug" else "Release";
+
+  meta = with lib; {
+    homepage = "https://phonon.kde.org/";
+    description = "GStreamer backend for Phonon";
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ ttuegel ];
+    license = licenses.lgpl21;
+  };
```
Comment (bjornfor) [inline pkgs/development/libraries/phonon/backends/gstreamer.nix:69]: Ah, I see you _moved_ meta. IMHO it's a good idea to keep commits strictly on topic and not do unrelated refactoring.

File: pkgs/development/libraries/phonon/backends/gstreamer.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/phonon/backends/gstreamer.nix:17]: moved to #245102

**Comment by SuperSandro2000** [comment]
removed those again :)

**Review by bjornfor — APPROVED** [review]

**Comment by AndersonTorres** [comment]
Fix the merge conflicts.

**Comment by Artturin** [comment]
Rebased

**Comment by AndersonTorres** [comment]
Let's go!
